feat(clickup-cdk): adding datadog-service-catalog options [CLK-257861] (#189)
time-loop/clickup-projen: docs/datadog-service-catalog/README.md

 
 // import { javascript } from 'projen';
 const project = new clickupCdk.ClickUpCdkTypeScriptApp({
@@ -30,6 +30,37 @@ const project = new clickupCdk.ClickUpCdkTypeScriptApp({
   renovatebotOptions: {
     scheduleInterval: ['before 1am on Monday'],
   },
+
+  serviceCatalogOptions: {
+    serviceInfo: [
+      {
+        description: 'infratest service used to validate new features and functionality',
+        tier: 'lower',
+        lifecycle: 'testing',
+        team: 'cloud platform',
+      },
+    ],
+    squadContacts: [
+      {
+        name: 'Contact name',
+        type: datadogServiceCatalog.SquadContactType.EMAIL,
+        contact: 'contacttest@clickup.com',
+      },
+    ],
+    squadLinks: [
+      {
+        name: 'link test',
+        type: datadogServiceCatalog.SquadLinkType.OTHER,
+        url: 'https://test.clickup.com',
+      },
+      {
+        name: 'staging link test',
+        type: datadogServiceCatalog.SquadLinkType.OTHER,
+        url: 'https://staging.clickup.com',
+      },
+    ],
+    serviceTags: { tag1: 'tag1-test', tag2: 'tag2-test' },
+  },
 });
```
> Note: Validate the `serviceCatalogOptions` block definition section. 

> **Get this committed then run `npx projen` to update the `.github/workflowa/release.yml` file**

Reviewer: ```suggestion
**Get this committed then run `npx projen` to update the `.github/workflows/release.yml` file**
```

Author: Updated [58d928a](https://github.com/time-loop/clickup-projen/pull/189/commits/58d928ac145ed541fe4764defad916ee7d1dc8b0)

### serviceCatalogOptions block definition

| field                      | description                                                                                           | required | default                                          |
| -------------------------- | ----------------------------------------------------------------------------------------------------- | -------- | ------------------------------------------------ |
| serviceCatalogOptions{}    | parent configuration block to enable                                                                  | No       | `undefined`                                      |
| serviceInfo[]              | Information about the service that will be published to the service catalog                           | No       |                                                  |

> | serviceInfo[].serviceName  | The name of the service. This must be unique across all services.                                     | No       | `project.name`                                   |

Reviewer: What happens when the `serviceName` is not unique? What will a developer see? How do they troubleshoot? Please add a section to the README.md about this.

Author: updated [e1d11a9](https://github.com/time-loop/clickup-projen/pull/189/commits/e1d11a9c335d65e681a829ae5f31dbe56b08024c)

| -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------- | -------- | ------------------------------------------------ |
| serviceCatalogOptions{}    | parent configuration block to enable                                                                                               | No       | `undefined`                                      |
| serviceInfo[]              | Information about the service that will be published to the service catalog                                                        | No       |                                                  |
| serviceInfo[].serviceName  | The name of the service. This must be unique across all services. (this value could override existing Service Catalog information) | No       | `project.name`                                   |
| serviceInfo[].description  | Some details on what this service does                                                                                             | No       | Not Provided                                     |

> | serviceInfo[].application  | The application/product that this service assists                                                                                  | No       | `clickup`                                        |

Reviewer: ```suggestion
| serviceInfo[].application  | The application/product that this service assists                                                                                  | No       | Not Provided                                        |
```

> | serviceInfo[].description  | Some details on what this service does                                                                                             | No       | Not Provided                                     |
> | serviceInfo[].application  | The application/product that this service assists                                                                                  | No       | Not Provided                                        |
> | serviceInfo[].team         | Which squad owns this service                                                                                                      | No       |                                                  |

Reviewer: ```suggestion
| serviceInfo[].description  | Some details on what this service does                                                                                             | No       | Not Provided                                     |
| serviceInfo[].application  | The application/product that this service assists                                                                                  | No       | Not Provided                                     |
| serviceInfo[].team         | Which squad owns this service                                                                                                      | No       |                                                  |
```

| links[]                    | A list of links associated with the service.                                                                                       | No       | `undefined`                                      |
| links[].name               | The name of the link                                                                                                               | Yes      |                                                  |
| links[].type               | The type for the link. Acceptable values are: `doc`, `runbook`, `repo`, `dashboard` and `other`                                    | Yes      |                                                  |
| links[].url                | The URL of the link.                                                                                                               | Yes      |                                                  |
| serviceTags{}              | The list of tags that are associated with the service. Record<string, string>                                                      | No       | `project`, `release`, `version` and `actor` tags |

### Configure multiple services in the same project.

> If your `cdk-project` is managing multiple services, you can include all the related information in the **serviceInfo[]** array as follows:

Reviewer: ```suggestion
If your `my-repo-cdk` is managing multiple services, you can include all the related information in the **serviceInfo[]** array as follows:
```

```diff
@@ -1,5 +1,5 @@
 import { core } from '@time-loop/cdk-library';
+import { clickupCdk, datadogServiceCatalog } from '@time-loop/clickup-projen';

 // import { javascript } from 'projen';
 const project = new clickupCdk.ClickUpCdkTypeScriptApp({
@@ -30,6 +30,37 @@ const project = new clickupCdk.ClickUpCdkTypeScriptApp({
   renovatebotOptions: {
     scheduleInterval: ['before 1am on Monday'],
   },
+
+  serviceCatalogOptions: {
+    serviceInfo: [
+      {
+        serviceName: 'service-name-1',
+        description: 'description of service-name-1',

> +        application: 'application-name' #default values clickup',

Reviewer: ```suggestion
+        application: 'application-name',
```

+        tier: 'lower',
+        lifecycle: 'testing',
+        team: 'cloud platform',

> +        pagerdutyUrl: 'https://pagerduty.url.com', #pager duty url

Reviewer: ```suggestion
+        pagerdutyUrl: 'https://pagerduty.url.com', // pager duty url
```

+      },
+      {
+        serviceName: 'service-name-2',
+        description: 'description of service-name-2',

> +        application: 'application-name' #default values clickup',

Reviewer: ```suggestion
+        application: 'application-name',
```

+        tier: 'critical',
+        lifecycle: 'developing',
+        team: 'other team',

> +        pagerdutyUrl: 'https://pagerduty.url.com', #pager duty url

Reviewer: ```suggestion
+        pagerdutyUrl: 'https://pagerduty.url.com',
```
